pom.xml

```diff
@@ -1,56 +1,61 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0"
-	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-	<modelVersion>4.0.0</modelVersion>
-	<groupId>junit45</groupId>
-	<artifactId>junit45</artifactId>
-	<version>0.0.1-SNAPSHOT</version>
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>junit45</groupId>
+  <artifactId>junit45</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
 
-	<properties>
-		<maven.compiler.source>1.8</maven.compiler.source>
-		<maven.compiler.target>1.8</maven.compiler.target>
-	</properties>
-	
-	<dependencies>	
-		<dependency>
-			<groupId>org.junit.jupiter</groupId>
-			<artifactId>junit-jupiter-engine</artifactId>
-			<version>5.7.0</version>
-			<scope>test</scope>
-		</dependency>
-		<dependency>
-			<groupId>org.junit.vintage</groupId>
-			<artifactId>junit-vintage-engine</artifactId>
-			<version>5.7.0</version>
-		</dependency>
-		<dependency>
-			<groupId>junit</groupId>
-			<artifactId>junit</artifactId>
-			<version>4.12</version>
-			<scope>test</scope>
-		</dependency>
-		<dependency>
-            <groupId>org.junit.jupiter</groupId>
-            <artifactId>junit-jupiter-api</artifactId>
-            <version>5.7.0</version>
-            <scope>test</scope>
-        </dependency>
-		
-		<!-- JUnit 5 tests get ignored after adding below dependency. Commenting the dependency will cause compilation error in JUnitPlatformClassDemoTest (due to JUnitPlatform class) -->		
-		<dependency>
-			<groupId>org.junit.platform</groupId>
-			<artifactId>junit-platform-runner</artifactId>
-			<version>1.7.0</version>
-			<scope>test</scope>
-		</dependency> 
```

Comment:
This would need to be retained to resolve compilation errors from JUnitPlatformClassDemoTest (when using @RunWith(JUnitPlatform.class)).

Reply:
Why do you need JUnitPlatform... Only needed if you are on a IDE/Tool which does not support JUnit Jupiter platform which is not the case for the majority of build tools and IDE's. Running on JUnit 4 can simply being done by using the JUnit 4 annotations that's it..

Reply:
I need it to be able to use PowerMock. When using PowerMock, I need to annotate the test class with @PowerMockRunnerDelegate(JUnitPlatform) to instruct the PowerMock runner to delegate to the runner that runs the JUnit platform based test. This is required to make sure test executor executes tests with JUnit 4.
The test case I created here is a simple one to show the problem.

Reply:
I would say using PowerMock is a big code smell in general or design problem but of course if you have to...then you have to go with it....

Reply:
Yes, PowerMock comes with its own challenges, but this is some legacy code and we cannot change it overnight. Meanwhile, I found something interesting. In the original pom version if I just change the maven-surefire-plugin version to 3.0.0-M4, then everything works fine (i.e. even with junit-platform-runner dependency, Maven is able to detect/execute Junit 5 test cases). Probably a bug in maven-surefire-plugin 3.0.0-M5 ?